### DISCUSSION TITLE
Can tracing be accomplished without using AWS X-Ray SDK and its dependancy on botocore (without using layer/SAR)?

### DISCUSSION THREAD

**Discussion by gthorp91** [aws-powertools/powertools-lambda-python#2269] [General]
When updating power tools to version >=2.6.0 the removal of the dependancies on boto3/botocore was a wonderful change, however when including tracing, AWS X-Ray SDK brings along a dependancy on botocore.

I have avoided currently this by using the powertools layer, but it would be nice to have the option of including the package without the additional file size that botocore brings along with it. It does appear that this issue has been raised in January this year on that repo but is currently on the backlog (https://github.com/aws/aws-xray-sdk-python/issues/378) - Powertools is even suggested as the model to follow! 

Is it possible in the meantime to have tracing enabled without using AWS X-Ray SDK?

**Comment by leandrodamascena** [comment]
Hi @gthorp91, thanks for bringing up this question, performance is something we always look to optimize in Powertools.

This question of installing AWS X-Ray SDK without its transitive dependencies, I don't know if we have much to do, because the one who solves the dependencies is the AWS X-Ray SDK package, and if we remove it, I don't know if it breaks anything, especially in local test environments. We created this model of splitting additional dependencies (pydantic, boto3, aws-xray-sdk and others) to allow the user to install only what they want and not have a very large deployment package.

> Is it possible in the meantime to have tracing enabled without using AWS X-Ray SDK?

Currently, no, it's not possible because our Tracer utility is a thin wrapper for the AWS X-Ray Python SDK. But the good news is that we already have a [RFC](https://github.com/awslabs/aws-lambda-powertools-python/issues/2030) to add external tracer providers to our Tracer Utility. It would be great if you could help us by commenting on this RFC, bringing your needs and experience with other Tracer providers.

Thank you
